Analyzer improvements (#18)
michalbali256/HlasmPlugin-docs: documentation/processing.tex

\label{chap:process}
Processing manager is a major component in the processing of a HLASM source file. It decides which stream of statements is to be processed and how statements are going to be processed. It contains components responsible for instruction interpretation as well as instruction format validation. 

> Nature of the HLASM source interpretation requires that the execution of various parsers and code generators runs in parallel. Processing manager performs this by maintaining 2 sets of active generators ("providers") and consumers (parsers, "processors") and executing them on demand, in an interleaved manner.

Reviewer: ```suggestion
Nature of the HLASM source interpretation requires that various parsers and code generators interlace to implement semantics of all instructions. Processing manager performs this by maintaining 2 sets of active generators ("providers") and consumers (parsers, "processors") and executing them on demand, in an interleaved manner.
```

\begin{enumerate}
	\item When a processor finishes its work, the next processor is selected from the array.

> 	\item When a provider finishes --- before the next provider is selected from the array --- manager checks whether it triggers the termination of the current processor as well (see \emph{terminal condition} in \cref{lab06:sect_proc}). Only if true, perform rule 1.

Reviewer: Stale z toho podla mna nie je jasne co sa stane ked je to false.
```suggestion
	\item When a provider finishes --- before the next provider is selected from the array --- manager checks whether it triggers the termination of the current processor as well (see \emph{terminal condition} in \cref{lab06:sect_proc}). Only if true, perform rule 1, otherwise the current processor stays active.
```

\end{enumerate}

\subsection{Initial state of manager}
\label{lab06:lib_data}

> During initialization, the manager can set various statement providers and processors as a default. It is very important as it determines the way how the source is processed. The manager determines this from \emph{library data} passed by analyzer.

Reviewer: ```suggestion
During initialization, the manager sets various statement providers and processors as a default. It is very important as it determines the way how the source is processed. The manager determines this from \emph{library data} passed by analyzer.
```

Author: preco?
teraz to znie akoby ich vedel viac setnut ako default naraz

Reviewer: z tohto pohladu tie 2 vety hovoria podla mna to iste

These providers are activated when COPY instruction copies a file into the source code or when a macro is visited, respectively. They provide a sequence of statements to an arbitrary processor until all statements from the copy or macro definition are provided. After that, if there is no nested invocation, a provider with lower priority is selected.

> As HLASM allows macro recursion, the reader may think that macro definition provider may never finish so the main loop may never break. For this purpose, HLASM language itself has restriction for the level of nested macro invocation depending on the complexity of nested macros. We set the limit to 100 as it suffices in all tested code.

Reviewer: "the reader may think" je podla mna take divne
```suggestion
To avoid infinite macro recursion, HLASM language itself has a restriction for the level of nested macro invocation depending on the complexity of nested macros. We set the limit to 100 as it suffices in all tested code.
```
